### ISSUE TITLE
Module loading sequence hang with OrdinaryDiffEq.jl

### ISSUE THREAD

**Issue by vram0gh** [PerezHz/TaylorIntegration.jl#160]
When using `OrdinaryDiffEq.jl` and `TaylorIntegration.jl` together, the sequence of using statements always has to be:

```julia
using OrdinaryDiffEq
using TaylorIntegration
```

in that order, otherwise a hang occurs. For example, this hangs:

```julia
using TaylorIntegration
using OrdinaryDiffEq
```

and so does this even though the using statements occur in sister modules:

```julia
module ModTIOnly
  using TaylorIntegration
end

module ModODEjlOnly
  using OrdinaryDiffEq
end
```

I originally opened this issue in the OrdinaryDiffEq.jl repo, https://github.com/SciML/OrdinaryDiffEq.jl/issues/1931 using the logic that since the hang occurs when loading `OrdinaryDiffeq` , but not the other way around, the proximate cause was in `OrdinaryDiffEq` and not `TaylorIntegration`, but it was closed there indicating the issue should be filed with this package.

Versions: Julia 1.8.5 on Linux/x86-64, OrdinaryDiffEq v6.50.0 , TaylorIntegration v0.11.1

**Comment by lbenet** [comment]
Thanks for reporting here this problem. I can indeed reproduce this.

I'm not sure why it happens. Could it be that we use `Requires.jl` and check if DiffEqBase is loaded, but not if OrdinaryDiffEq is? Should we add both as direct dependencies? Any clues?

cc @ChrisRackauckas

**Comment by ChrisRackauckas** [comment]
If you're using anything from OrdinaryDiffEq.jl then you need to `@requires` it. Though I'd highly recommend changing from Requires to the newer package extensions.

**Comment by lbenet** [comment]
> If you're using anything from OrdinaryDiffEq.jl then you need to @requires it. Though I'd highly recommend changing from Requires to the newer package extensions.

That would only work for Julia >= v1.9, am I right? What about v1.8?

**Comment by ChrisRackauckas** [comment]
Some issues with Requires are just not solvable, so I wouldn't worry too much.

**Comment by lbenet** [comment]
@vram0gh This is fixed by #161, though the fix does not exploit the new pkg extensions...
